modules/Hoot/tools/selectBbox.js

```diff
@@ -15,17 +15,17 @@ import { d3combobox }             from '../../lib/hoot/d3.combobox';
 import { geoExtent as GeoExtent } from '../../geo';
 
 export default class SelectBbox extends EventEmitter {
-    constructor( context ) {
+    constructor( context, predefinedData ) {
         super();
 
-        this.context = context;
-
-        this.minlon         = null;
-        this.minlat         = null;
-        this.maxlon         = null;
-        this.maxlat         = null;
-        this.bboxSelectType = 'visualExtent';
+        this.context        = predefinedData && predefinedData.context ? predefinedData.context : context;
+        this.minlon         = predefinedData && predefinedData.minlon ? parseFloat(predefinedData.minlon) : null;
+        this.minlat         = predefinedData && predefinedData.minlat ? parseFloat(predefinedData.minlat) : null;
+        this.maxlon         = predefinedData && predefinedData.maxlon ? parseFloat(predefinedData.maxlon) : null;
+        this.maxlat         = predefinedData && predefinedData.maxlat ? parseFloat(predefinedData.maxlat) : null;
+        this.bboxSelectType = predefinedData && predefinedData.bboxSelectType ? predefinedData.bboxSelectType : 'visualExtent';
         this.operationName  = '';
+        this.selectedBoundOption = predefinedData && predefinedData.selectedBoundOption ? predefinedData.selectedBoundOption : 'Visual Extent';
```

Comment:
this stuff is pretty ugly but couldnt figure out a really nice way. The other option was to just set the defaults here and maybe in a helper function replace the defaults with the 'predefinedData' if it was defined